controllers/sriov-controller/dpapi-controller.go

```diff
@@ -244,24 +232,89 @@ func (s *serviceInstanceController) ListAndWatch(e *pluginapi.Empty, d pluginapi
+	// the network service.
+	vfioDevs := []vfioConfig{}
+	// Bulding per network service key for Env variable, it will point to the network service configuration
+	// file.
+	networkServiceName := strings.ToLower(strings.Split(s.networkServiceName, "/")[1])
+	networkServiceName = strings.Replace(networkServiceName, "-", "_", -1)
+	if strings.HasPrefix(networkServiceName, "sriov_") {
+		networkServiceName = strings.Split(networkServiceName, "sriov_")[1]
+	}
+	key := "NSM_VFS_" + networkServiceName
+	// Creating config file which will be passed to the POD. The config file name is composed
+	// from Network Service name + vfio group ID taken from the first device in Allocate Request.
+	// Example: If Network Service is vlan10 and AllocateRequest has /dev/vfio/67, then the config
+	// file name will be vlan10_67.json.
+	// If a file with the same name already exists, os.Create will truncate it and previous content will be lost.
+	_, groupID := path.Split(reqs.ContainerRequests[0].DevicesIDs[0])
+	configFileName := fmt.Sprintf("%s_%s.json", networkServiceName, groupID)
+	configFile, err := os.Create(path.Join(containersConfigPath, configFileName))
+	if err != nil {
+		return nil, fmt.Errorf("fail to create network services config file with error: %+v", err)
+	}
+	defer configFile.Close()
 	for _, req := range reqs.ContainerRequests {
 		response := pluginapi.ContainerAllocateResponse{
 			Devices: []*pluginapi.DeviceSpec{},
+			// Adding env variable for requested network service, the key is composed as "NSM_VFS_"+ {network service name}
+			// excluding organization prefix.
+			// Environment variable points to the location of the network service specific configuration file
+			Envs: map[string]string{
+				key: path.Join(containerConfigFilePath, configFileName),
+			},
+			Mounts: []*pluginapi.Mount{
+				&pluginapi.Mount{
+					// Adding this specific network service configuration file into the container
+					ContainerPath: path.Join(containerConfigFilePath, configFileName),
+					HostPath:      configFile.Name(),
+					ReadOnly:      true,
+				},
+			},
 		}
 		for _, id := range req.DevicesIDs {
 			deviceSpec := pluginapi.DeviceSpec{}
 			logrus.Infof("Allocation request for device: %s", id)
 			if !s.checkVF(id) {
-				return nil, fmt.Errorf("invalid allocation request: unknown device: %s", id)
+				return nil, fmt.Errorf("allocation request failure, unknown device: %s", id)
 			}
 			deviceSpec.HostPath = id
 			deviceSpec.ContainerPath = id
 			deviceSpec.Permissions = "rw"
 			response.Devices = append(response.Devices, &deviceSpec)
+			// Getting vfio device specific specifications and storing it in the slice. The slice
+			// will be marshalled into json and passed to requesting POD as a mount.
+			vfioDev, err := s.getVFIODevSpecs(id)
+			if err != nil {
+				return nil, fmt.Errorf("allocation request failure, unable to get device %s specs with error: %+v", id, err)
```

Comment:
Same here, maybe a soft failure and continue processing?

Reply:
I would still prefer a consistency, if POD requested these resources and did not get ALL, then it should fail to start as behaviour of application might be unpredictable.